lib/api/core/plugins/pluginsManager.js

```diff
@@ -825,13 +830,19 @@ class PluginsManager {
 /**
  * Emit event
  *
- * @param {EventEmitter} emitter
+ * @param {object} hooks
  * @param {string} event
  * @param {*} data
  * @returns {Promise}
  */
-function triggerHooks(emitter, event, data) {
-  emitter.emit(event, data);
+function triggerHooks(hooks, event, data) {
+  const preparedHooks = getPreparedListeners(hooks, event);
+
+  if (preparedHooks) {
+    preparedHooks.forEach(hook => {
+      hook(data);
```

Comment:
I'm not sure but what if the hook function provided by the developer throw an error ? Can't we have a unhandled rejection here ?

Reply:
more importantly: this bypasses Kuzzle as an event emitter, which is not the purpose of hooks